dask/dataframe/io/tests/test_json.py

```diff
@@ -13,18 +13,97 @@
 ddf = dd.from_pandas(df, npartitions=2)
 
 
+@pytest.mark.parametrize("orient", ["split", "records", "index", "columns", "values"])
+def test_read_json_with_path_column(orient):
+    with tmpfile("json") as f:
+        df.to_json(f, orient=orient, lines=False)
+        actual = dd.read_json(f, orient=orient, lines=False, include_path_column=True)
+        actual_pd = pd.read_json(f, orient=orient, lines=False)
+        # The default column name when include_path_colum is True is "path"
+        # The paths on Windows are converted to forward slash somewhere in the file
+        # reading chain in Dask, so we have to do the same here.
+        actual_pd["path"] = pd.Series(
+            (f.replace(os.sep, "/"),) * len(actual_pd), dtype="category"
+        )
+        assert actual.path.dtype == "category"
+        assert_eq(actual, actual_pd)
+
+
+def test_read_json_path_column_with_duplicate_name_is_error():
+    with tmpfile("json") as f:
+        df.to_json(f, orient="records", lines=False)
+        with pytest.raises(ValueError, match="Files already contain"):
+            dd.read_json(f, orient="records", lines=False, include_path_column="x")
+
+
+def test_read_json_with_path_converter():
+    path_column_name = "filenames"
+
+    def path_converter(x):
+        return "asdf.json"
+
+    with tmpfile("json") as f:
+        df.to_json(f, orient="records", lines=False)
+        actual = dd.read_json(
+            f,
+            orient="records",
+            lines=False,
+            include_path_column=path_column_name,
+            path_converter=path_converter,
+        )
+        actual_pd = pd.read_json(f, orient="records", lines=False)
+        actual_pd[path_column_name] = pd.Series(
+            (path_converter(f),) * len(actual_pd), dtype="category"
+        )
+        assert_eq(actual, actual_pd)
+
+
+def test_read_orient_not_records_and_lines():
+    with pytest.raises(ValueError, match="Line-delimited JSON"):
+        dd.read_json("nofile.json", orient="split", lines=True)
+
+
+def test_write_orient_not_records_and_lines():
+    with tmpfile("json") as f:
+        with pytest.raises(ValueError, match="Line-delimited JSON"):
+            dd.to_json(ddf, f, orient="split", lines=True)
+
+
+@pytest.mark.parametrize("blocksize", [5, 15, 33, 200, 90000])
+def test_read_json_multiple_files_with_path_column(blocksize, tmpdir):
```

Comment:
It looks like this needs the same path formatting you applied to `test_read_json_with_path_column`

@martindurant do you have any insight into why `pandas.to_json` and `dask.dataframe.read_json` are using different path separators on windows (i.e. pandas outputs files with `\` and Dask reads them in with a `/`)?